src/main/java/org/broadinstitute/consent/http/resources/DataSetResource.java

```diff
@@ -112,9 +113,14 @@ public Response createDataset(@Auth AuthUser user, @Context UriInfo info, String
         }
         User dacUser = userService.findUserByEmail(user.getGoogleUser().getEmail());
         Integer userId = dacUser.getDacUserId();
-        DataSet createdDataset = datasetService.createDataset(inputDataset, name, userId);
-        URI uri = info.getRequestUriBuilder().replacePath("api/dataset/{datasetId}").build(createdDataset.getDataSetId());
-        return Response.created(uri).entity(createdDataset).build();
+        DataSetDTO createdDataset = datasetService.createDataset(inputDataset, name, userId);
+        Consent createdConsent = datasetService.createConsentForDataset(createdDataset);
```

Comment:
This can throw an `IllegalArgumentException` so we should handle that case here.

Reply:
Sure! In the event that a Consent cannot be created, should we be rolling back the new Dataset as well?

Reply:
Excellent point, yes, the whole operation should fail in that case. If there's anything we can do to get the reason behind the failure out to the UI, that would be helpful.